### DISCUSSION TITLE
Can't access filesystem in express

### DISCUSSION THREAD

**Discussion by VelrinBlack** [vercel/vercel#5609] [Help]
Hello,
I'm trying to deploy my `express.js` app to vercel. In one of the files, I make:
```js
fs.readFileSync('src/mail.ejs', 'utf8')
```
On `localhost` it works well, but after deployment, I can see `Error: ENOENT: no such file or directory, open 'src/mail.ejs`.
Here's my `vercel.json`:
```json
{
  "version": 2,
  "builds": [
    {
      "src": "./server.js",
      "use": "@now/node-server"
    }
  ],
  "routes": [
    { "handle": "filesystem" },
    {
      "src": "/.*",
      "dest": "server.js"
    }
  ]
}
```
Thanks in advance for the help

**Comment by danchoivlado** [comment]
Its generally considered an anti-pattern to write files from your Serverless Functions because you can't guarantee the next invocation is the same function so the file might be missing.

The best solution is to use database.
